### PR TITLE
Move explain to standalone task.

### LINKED ISSUE #179: Add explanations for warnings with default/missing explanations.
- [ ] `Dialyxir.Warnings.AppCall`
- [ ] `Dialyxir.Warnings.BinaryConstruction`
- [ ] `Dialyxir.Warnings.CallWithOpaque`
- [ ] `Dialyxir.Warnings.CallWithoutOpaque`
- [ ] `Dialyxir.Warnings.ContractDiff`
- [ ] `Dialyxir.Warnings.FuncionApplicationArguments`
- [ ] `Dialyxir.Warnings.ImproperListConstruction`
- [ ] `Dialyxir.Warnings.OpaqueEquality`
- [ ] `Dialyxir.Warnings.OpaqeGuard`
- [ ] `Dialyxir.Warnings.OpaqueMatch`
- [ ] `Dialyxir.Warnings.OpaqueNonequality`
- [ ] `Dialyxir.Warnings.OpaqueTypeTest`
- [ ] `Dialyxir.Warnings.RaceCondition`
- [ ] `Dialyxir.Warnings.RecordConstruction` (likely not possible)
- [ ] `Dialyxir.Warnings.RecordMatching`   (likely not possible)
- [ ] `Dialyxir.Warnings.UnknownBehaviour`
- [ ] `Dialyxir.Warnings.UnknownFunction`

### PR DESCRIPTION
**Pull request by jeremyjh**
resolves #179

### CODE REVIEW

File: lib/mix/tasks/dialyzer/explain.ex [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      {_, [], _} ->
+        list_warnings() |> Output.info()
+
+      _ ->
+        Mix.Task.run("help", ["dialyzer.explain"])
+    end
+  end
+
+  defp explanation_text(warning_name) do
+    warning = String.to_atom(warning_name)
```
Comment (asummers) [inline lib/mix/tasks/dialyzer/explain.ex]: I'm always a bit weary of `String.to_atom/1`. Should be fine here, though, I think.

File: lib/mix/tasks/dialyzer/explain.ex [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    module = Map.get(Dialyxir.Warnings.warnings(), warning)
+    (module && module.explain()) || "Uknown warning named: #{warning_name}"
```
Comment (asummers) [inline lib/mix/tasks/dialyzer/explain.ex]: `explain/1` will always return a legit value, so this check is really 

```
if module do
  module.explain()
else
  "Unknown warning named: #{warning_name}"
end
```

Also typo on `Unknown`

Comment (asummers) [reply lib/mix/tasks/dialyzer/explain.ex]: Or if you REALLY want to get fancy:

```
case Map.get(Dialyxir.Warnings.warnings(), warning) do
  nil -> 
    "Unknown warning named: #{warning_name}"
  module ->
    module.explain()
end
```

Comment (jeremyjh) [reply lib/mix/tasks/dialyzer/explain.ex]: If the warning name isn't found in the map, `module` will be nil so the left branch will fail and it prints the unknown. Do you just mean the `if`/`case` is more readable?

Comment (asummers) [reply lib/mix/tasks/dialyzer/explain.ex]: Yeah I think it's a little more honest as to intent. They're all functionally the same.

File: lib/mix/tasks/dialyzer/explain.ex (hunk 1 above)
Comment (jeremyjh) [reply lib/mix/tasks/dialyzer/explain.ex]: Yeah its the best choice here, we can't leak.

**Comment by asummers** [comment]
Might be worth mentioning the new task in the README too

**Pull request merged**